cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceProgressUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+/**
+ * Utility class for handling progress fields of KafkaRebalance custom resource
+ */
+public class KafkaRebalanceProgressUtils {
+
+    private static final int SECONDS_IN_A_MINUTE = 60;
+
+    /**
+     * Estimates the number of minutes it will take an ongoing partition rebalance to complete.
+     *
+     * @param taskStartTime The time when the task started, in seconds since the Unix epoch.
+     * @param currentTime The current time, in seconds since the Unix epoch, at the moment of the method call.
+     * @param totalDataToMove The total amount of data that needs to be moved, in megabytes.
+     * @param finishedDataMovement The amount of data that has already been moved, in megabytes.
+     * @return The estimated time to completion in minutes.
+     * @throws IllegalArgumentException if:
+     *     - Any of the method argument values are negative.
+     *     - The value of taskStartTime is not greater than value of the currentTime.
+     * @throws ArithmeticException if:
+     *     - The elapsed time between `taskStartTime` and `currentTime` is zero, making rate calculation impossible.
+     *     - The data movement rate is zero, making the time to completion estimation impossible.
+     */
+    /* test */ static int estimateTimeToCompletionInMinutes(long taskStartTime,
+                                                        long currentTime,
+                                                        int totalDataToMove,
```

Comment:
Can we add the data units to these variable names:
```suggestion
                                                        int totalDataToMoveMB,
```

Reply:
Fixed.